saleor/warehouse/availability.py

```diff
@@ -273,34 +290,145 @@ def check_preorder_threshold_bulk(
         )
         for variant_id, channel_listings in variant_channels.items()
     }
+    return (
+        variants_channel_availability,
+        variants_global_allocations,
+        all_variants_channel_listings,
+        variant_channels,
+    )
```

Comment:
Maybe at least a named tuple?

Reply:
I have managed even two typed named tuple (based on `NamedTuple` from typing). Please note, that there is at least one more function that has 3+ unnamed tuple (checked only in `availbility.py`